docs/wallet/address-tagging.md

```diff
@@ -0,0 +1,125 @@
+---
+id: address-tagging
+title: Address and UTXO tagging
+---
+
+```scala mdoc:invisible
+import org.bitcoins.core.wallet.utxo._
+import org.bitcoins.core.protocol.BitcoinAddress
+import org.bitcoins.core.protocol.transaction._
+import org.bitcoins.core.wallet.fee._
+import org.bitcoins.core.currency._
+import org.bitcoins.wallet.models.AccountDb
+import org.bitcoins.wallet._
+
+val ExampleAddressTag = CustomAddressTag("name", "tagType")
+val exampleAddress = BitcoinAddress("1A1zP1eP5QGefi2DMPTfTL5SLmv7DivfNa").get
+val account: AccountDb = null
+val destinations = Vector(TransactionOutput(Satoshis.one, exampleAddress.scriptPubKey))
+case object wallet {
+    // need to define these functions so we don't have to implement every function
+    def fundRawTransaction(
+              destinations: Vector[TransactionOutput],
+              feeRate: FeeUnit,
+              fromTagOpt: Option[AddressTag],
+              markAsReserved: Boolean) = true
+
+    def getNewAddress(tags: Vector[AddressTag]) = true
+
+    def sendToAddress(
+          address: BitcoinAddress,
+          amount: CurrencyUnit,
+          feeRate: FeeUnit,
+          fromAccount: AccountDb,
+          newTags: Vector[AddressTag]) = true
+}
+```
+
+### Using AddressTags
+
+The Bitcoin-S wallet allows you to give addresses, and their associated utxos,
+a tag. These tags allow you to separate funds between utxos so you can query utxos,
+and spend from them, based off of an AddressTag. The system also allows you to create
+your own custom address tags, that will be enforced by the library.
```

Comment:
I'm a little confused, is it true that a given utxo can only have one tag?

Reply:
No utxos can have any number of tags, just sending is only from a single tag currently to keep it simpler.